internal/kldeth/txn.go

```diff
@@ -121,13 +123,37 @@ func NewContractDeployTxn(msg *kldmessages.DeployContract, signer TXSigner) (tx
 }
 
 // CallMethod performs eth_call to return data from the chain
-func CallMethod(ctx context.Context, rpc RPCClient, signer TXSigner, from, addr string, value json.Number, methodABI *abi.Method, msgParams []interface{}) (map[string]interface{}, error) {
+func CallMethod(ctx context.Context, rpc RPCClient, signer TXSigner, from, addr string, value json.Number, methodABI *abi.Method, msgParams []interface{}, blocknumber string) (map[string]interface{}, error) {
 	log.Debugf("Calling method: %+v %+v", methodABI, msgParams)
 	tx, err := buildTX(signer, from, addr, "", value, "", "", methodABI, msgParams)
 	if err != nil {
 		return nil, err
 	}
-	retBytes, err := tx.Call(ctx, rpc)
+	callOption := "latest"
+	// only allowed values are "latest", "", a number string "12345" or a hex number "0xab23"
+	// "latest" and "" (no kld-blocknumber given) are equivalent
+	if blocknumber != "" && blocknumber != "latest" {
+		isHex, _ := regexp.MatchString(`^0x[0-9a-fA-F]+$`, blocknumber)
+		if isHex || blocknumber == "earliest" || blocknumber == "pending" {
+			callOption = blocknumber
+		} else {
+			n := new(big.Int)
+	    n, ok := n.SetString(blocknumber, 10)
+	    if !ok {
+	    	log.Warnf("Failed to convert blocknumber to big integer: %s", blocknumber)
+	    	// when failed to convert, use "latest"
+	    } else {
+				bits := n.BitLen()
```

Comment:
Not sure I understand this bit. Can't we just use the `hexutil.Big` in the parameter? It's designed to do the serialization for us.

Reply:
since the end value we want is a hex string, once the input is detected as a hex it's used directly.

This section is for parsing integer strings into `big.Int`. Getting the bit length seems to be the best way to check for zero value